pyirf/irf/effective_area.py

```diff
@@ -48,3 +48,38 @@ def point_like_effective_area(selected_events, simulation_info, true_energy_bins
     hist_simulated = simulation_info.calculate_n_showers_per_energy(true_energy_bins)
 
     return effective_area(hist_selected["n"], hist_simulated, area)
+
+
+def effective_area_energy_fov(
+    selected_events, simulation_info, true_energy_bins, fov_offset_bins
+):
+    """
+    Calculate effective area in bins of true energy and field of view offset.
+
+    Parameters
+    ----------
+    selected_events: astropy.table.QTable
+        DL2 events table, required columns for this function: `true_energy`.
+    simulation_info: pyirf.simulations.SimulatedEventsInfo
+        The overall statistics of the simulated events
+    true_energy_bins: astropy.units.Quantity[energy]
+        The true energy bin edges in which to calculate effective area.
+    fov_offset_bins: astropy.units.Quantity[angle]
+        The field of view radial bin edges in which to calculate effective area.
+    """
+    area = np.pi * simulation_info.max_impact ** 2
+
+    hist_simulated = simulation_info.calculate_n_showers_per_energy_and_fov(
+        true_energy_bins, fov_offset_bins
+    )
+
+    hist_selected, _, _ = np.histogram2d(
+        selected_events["true_energy"].to_value(u.TeV),
+        selected_events["source_fov_offset"].to_value(u.deg),
```

Comment:
couldn't the histogram be computed with units?

Reply:
It could yes, but normally it's faster to strip away units if possible. Here it is easy, since we know what units to convert too:


```
In [1]: import astropy.units as u
In [2]: import numpy as np
In [3]: energies = np.random.normal(100, 10, 10000) * u.TeV
In [4]: offsets =  np.abs(np.random.normal(0, 0.1, 10000)) * u.deg
In [5]: energy_bins = np.array([80, 90, 100, 110, 120]) * u.TeV
In [6]: offset_bins = np.arange(0, 1.0, 0.1) * u.deg

In [7]: %%timeit
   ...: np.histogram2d(energies, offsets, bins=[energy_bins, offset_bins])

410 µs ± 8.02 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [8]: %%timeit
   ...: np.histogram2d(energies.to_value(u.TeV), offsets.to_value(u.deg), bins=[energy_bins.to_value(u.TeV), offset_bins.to_value(u.deg)])

379 µs ± 6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Admittedly, the difference is not large here